addons/mail/static/src/models/partner/partner.js

```diff
@@ -235,6 +278,13 @@ function factory(dependencies) {
         failureNotifications: one2many('mail.notification', {
             related: 'messagesAsAuthor.failureNotifications',
         }),
+        /**
+         * Whether an attempt was already made to fetch the user corresponding
+         * to this partner. This prevents doing the same RPC multiple times.
+         */
+        hasCheckedUser: attr({
+            default: false,
+        }),
```

Comment:
just a remark that this design is weak against having checked a user and someone changed relation between partner and user at some point.

...Not that we do that every day, but just pointing out a limitation with this.

Reply:
Agree with limitation, solutions exist if it ever becomes a problem:
- save last check datetime instead of boolean to allow the check again after some time
- at least for the person doing the change and the person itself (if applicable): ensure proper partner is being linked at the same time, so that no other check is necessary
- unlimited resource world: send on bus every time user or partner is created or their relation modified

Reply:
I'm pretty sure the issue will happen actually, probably not much with partner/user, but especially when linking employee to user this will happen.

I guess that's fine for now, at worse f5 will fix it... I wouldn't want to delay the PR further for this, especially since it was already an issue with the previously existing "many2one avatar" caching.

Reply:
As I said, it was mainly a remark, not a request for change in this PR.

It's important to have in mind the consequences of simplifications we make in implementing some features, especially limitations — like the one I just pointed out — that could become a big issue. Yes, F5 "fixes" the problem, but this is like having an big issue on OS and you can just "fix" it by rebooting your machine...

Reply:
task-2327662 for reference, not sure if we'll ever do it or not